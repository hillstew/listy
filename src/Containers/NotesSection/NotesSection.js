import React from 'react';
import Note from '../Note/Note';
import { connect } from 'react-redux';
import PropTypes from "prop-types"


const NotesSection = ({ notes }) => {
  return (
    <div id='notes-wrapper'>
    {notes.map(note => <Note note={note} key={note.id}/>)}
    </div>
  )
}

const mapStateToProps = (state) => ({
  notes: state.notes,
});

export default connect(mapStateToProps)(NotesSection);

NotesSection.propTypes = {
  notes: PropTypes.array.isRequired,
}