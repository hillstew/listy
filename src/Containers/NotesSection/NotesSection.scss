#notes-wrapper {
  padding-top: 240px;
  width: 90%;
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(280px, 280px));
  grid-gap: 30px;
  margin: 0 auto 40px auto;
  justify-content: start;
}