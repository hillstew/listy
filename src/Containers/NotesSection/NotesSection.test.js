import React from 'react';
import { NotesSection, mapStateToProps } from './NotesSection';
import { shallow } from 'enzyme';

describe('NotesSection', () => {
  describe('NotesSection component', () => {
    let wrapper;
    let mockNotes;

    beforeEach(() => {
      mockNotes = [
        { id: '1', title: 'title1', issues: [] },
        { id: '1', title: 'title2', issues: [] },
      ];
      wrapper = shallow(
        <NotesSection 
          notes={mockNotes}
        />
      )
    });

    it('should match the correct snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('mapStateToProps', () => {
    it('should return redux state as props', () => {
      const mockState = {
        notes: [],
        error: ''
      }

      const expected = {
        notes: []
      }

      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expected);
    });
  });
});