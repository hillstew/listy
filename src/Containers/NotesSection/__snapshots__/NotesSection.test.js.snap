// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotesSection NotesSection component should match the correct snapshot 1`] = `
<div
  id="notes-wrapper"
>
  <Connect(Note)
    key="1"
    note={
      Object {
        "id": "1",
        "issues": Array [],
        "title": "title1",
      }
    }
  />
  <Connect(Note)
    key="1"
    note={
      Object {
        "id": "1",
        "issues": Array [],
        "title": "title2",
      }
    }
  />
</div>
`;
